luigi/contrib/batch.py

```diff
@@ -0,0 +1,213 @@
+
+        return response['jobs'][0]['status']
+
+    def get_logs(self, log_stream_name, get_last=50):
+        """Retrieve log stream from CloudWatch"""
+        response = self._log_client.get_log_events(
+            logGroupName='/aws/batch/job', 
+            logStreamName=log_stream_name,
+            startFromHead=False)
+        events = response['events']
+        return '\n'.join(e['message'] for e in events[-get_last:])
+
+    def submit_job(self, job_definition, parameters, job_name=None, queue=None):
+        """Wrap submit_job with useful defaults"""
+        if job_name is None:
+            job_name = _random_id()
+        response = self._client.submit_job(
+            jobName = job_name,
+            jobQueue = queue or self.get_active_queue(),
+            jobDefinition = job_definition,
+            parameters = parameters
+        )
+        return response['jobId']
+
+    def wait_on_job(self, job_id):
+        """Poll task status until STOPPED"""
+        
+        while True:
+            status = self.get_job_status(job_id)
+            if status == 'SUCCEEDED':
+                logger.info('Batch job {} SUCCEEDED'.format(job_id))
+                return True
+            elif status == 'FAILED':
+                # Raise and notify if job failed
+                jobs = self._client.describe_jobs(jobs=[job_id])['jobs']
+                job_str = json.dumps(jobs, indent=4)
+                logger.debug('Job details:\n' + job_str)
+                
+                log_stream_name = jobs[0]['attempts'][0]['container']['logStreamName']
+                logs = self.get_logs(log_stream_name)
+                raise BatchJobException('Job {} failed: {}'.format(
+                    job_id, logs))
+
+            time.sleep(POLL_TIME)
+            logger.debug('Batch job status for job {0}: {1}'.format(
+                job_id, status))
+
+    def register_job_definition(self, json_fpath):
+        """Register a job definition with AWS Batch, using a JSON"""
+        with open(json_fpath) as f:
+            job_def = json.load(f)
+        response = self._client.register_job_definition(**job_def)
+        status_code = response['ResponseMetadata']['HTTPStatusCode']
+        if status_code != 200:
+            msg = 'Register job definition request received status code {0}:\n{1}'
+            raise Exception(msg.format(status_code, response))
+        return response
+
+
+class BatchTask(luigi.Task):
+
+    """
+    Base class for an Amazon Batch job
+
+    Amazon Batch requires you to register "job definitions", which are JSON 
+    descriptions for how to issue the ``docker run`` command. This Luigi Task 
+    requires a pre-registered Batch jobDefinition name passed as a Parameter
+
+    :param job_definition: name of pre-registered jobDefinition
+    :param job_name: name of specific job, for tracking in the queue and logs.
+
+    """
+    job_definition = luigi.Parameter()
+    job_name = luigi.Parameter(default=None)
```

Comment:
This can use the new `OptionalParameter` to avoid the String default is None warning (however, this feature has not yet been released to stable yet)